Source/WebCore/workers/service/context/ServiceWorkerThread.cpp

```diff
@@ -116,8 +116,18 @@ static WorkerParameters generateWorkerParameters(const ServiceWorkerContextData&
     };
 }
 
+static WorkerThreadStartMode threadStartModeFromSettings(const Settings::Values& settingsValues)
+{
+#if ENABLE(REMOTE_INSPECTOR)
+    if (settingsValues.developerExtrasEnabled)
```

Comment:
is there a way to also check for if the developer has enabled "Automatically Pause Connecting to ServiceWorkers"?

Reply:
Also, we want the debuggable to pause regardless of "Automatically Pause Connecting to ServiceWorkers" because a genuine auto-inspection needs to appear attached to the debuggable before it starts executing scripts, which means the debuggable should spawn assuming it will be auto-inspected unless proven otherwise.

Reply:
I have reverted back to always using `WaitForInspector` as the start mode. I'll find a better heuristic as a follow-up task (filed [here](https://bugs.webkit.org/show_bug.cgi?id=288552), with more explanation).